Add handling for governance metadata

Bankso opened the pull request:
Added option to provide governance-focused Study metadata as a Synapse Id. This is assumed to be a RecordSet, which is downloaded and parsed to annotations. Study DUO-related metadata is applied as annotations entity at Synapse Entity Id in the StudyProjectIdentifier column.

Bankso committed:
Add handling for governance metadata

Added option to provide governance-focused Study metadata as a Synapse Id. This is assumed to be a RecordSet, which is downloaded and parsed to annotations. Study DUO-related metadata is applied as annotations entity at Synapse Entity Id in the StudyProjectIdentifier column.

aditya-nath-sage commented on utils/table_to_annotations.py:
```diff
@@ -57,17 +60,35 @@ def get_args():
     """Collect columns from a Synapse table entity and return as a Dataframe."""
 
-    if type(cols) == list:
+    if type(cols) == list and is_record_set is False:
```
Do we want to add some error handling here? maybe a try-except block?

aditya-nath-sage commented on utils/table_to_annotations.py:
```diff
@@ -329,11 +362,67 @@ def main():
         "Data Use Codes",
     ]
 
+    ada_psi_study_columns = [
```
Possibly make this a constant outside of the script for easier editing?

aditya-nath-sage reviewed (approved):
LGTM!

Bankso committed:
Update description

Bankso committed:
Add option to pass files to remove list

Bankso committed:
Print correct dataset count

Bankso committed:
Add script to move files

Bash loop to run synapse mv for a set of file id/target id pairs

Bankso committed:
Create create_curation_task.py

Based on example in synapse client docs: https://python-docs.synapse.org/en/stable/guides/extensions/curator/metadata_curation/

Bankso committed:
Move to utils

Bankso committed:
Enable scripts and functions to be imported in utils

Bankso committed:
Update create_curation_task script

Add argparse and conversion of inputs to function args

Bankso committed:
Modify script for external use

Rename main and return a URI that can be used by Curator to bind a schema to a RecordSet or FileView

Bankso committed:
Standardize task naming

Bankso committed:
Support name parsing for Record Sets

Different separator and name pattern than JSON schemas

Bankso committed:
Allow RecordSets to be used as source for UNION tables

Added a function to identify record sets of a given data type, pull the content, create a new Synapse table for each record set in its source Synapse project, then create a materialized view for all the new tables. This can be configured to recapitulate the structure, schemas, etc. of existing synapse_storage_manifest_tables, so that Curator and RecordSets can be easily integrated into the CCKP database as needed

Bankso committed:
Allow an input sheet to be used for multiple tasks

Sheet should be a CSV with columns matching input flags: project, folder, data_type, record_desc, primary_keys, instructions, schema_uri, schema_path, task_type, version

Bankso committed:
Use source table for project ids

Bankso committed:
Update docs on schema naming and parsing

Bankso committed:
Allow single manifest or link to be used as input

Bankso committed:
Condense code

Bankso committed:
Add flag to pass version at command line

Bankso committed:
Update .gitignore

Bankso committed:
Create curation_tasks_TEMPLATE.csv

Bankso committed:
Update input handling

Bankso committed:
Parse inputs to strings

Bankso committed:
Create delete_curation_task.py

Bankso committed:
Create list_curation_tasks.py

Bankso committed:
Add basic documentation
